configure.ml

```diff
@@ -456,7 +456,7 @@ let coq_bin_annot_flag = if !prefs.bin_annot then "-bin-annot" else ""
 let coq_safe_string = "-safe-string"
 let coq_strict_sequence = "-strict-sequence"
 
-let cflags = "-Wall -Wno-unused -g -O2"
+let cflags = "-Wall -Wno-unused -g -O2 -fexcess-precision=standard -msse2 -mfpmath=sse"
```

Comment:
> What are Coq's supported architectures?

We could not find any indication in the documentation (README.md, INSTALL, sphinx doc).
Maybe is it time to add something. I don't know what would be the best place for this indication though. @Zimmi48 any idea?

> make sure that the SSE2 flags are only passed if the compiler is able to recognize them.

Done. The main incompatibility we are aware of is x87 which is the default for both OCaml and gcc on x86_32, leading to double roundings. The logic we just implemented in `configure.ml` is the following:
* test for SSE2 support to add corresponding flags to `CFLAGS`
* test for double rounding in OCaml
  * if no double rounding happens, fine
  * if double rounding happens and SSE2 is available, fine (it will be used, preventing double rounding)
  * otherwise, abort configure

In case miscompilation still happens, a runtime test (at the end of `kernel/float64.ml` could still catch it).

The code is in the following commit: https://github.com/coq/coq/pull/9867/commits/158561390aa496b42c2bf993629e39de77bdbe62 (we had to rebase for CI)

Currently, the test for SSE2 availability is done by checking the presence of `#define __SSE2_MATH__ 1` in the output of `cc -march=native -dM -E -msse2 -mfpmath=sse some_file.c` @MSoegtropIMC isn't it to Unix specific? what about Windows?

Reply:
If we need to mess with c flags it also needs to be done in the dune file btw

Reply:
> We could not find any indication in the documentation (README.md, INSTALL, sphinx doc).
> Maybe is it time to add something. I don't know what would be the best place for this indication though. @Zimmi48 any idea?

Sure it would seem like something sensible to document. I suggest doing it in another PR, and indeed `INSTALL` seems like the reasonable place for this. Note though how some stuff like the VM and native compute machinery have fallback mechanisms that make the use of Coq possible even when these features cannot be used.

Reply:
> these features

Which features do you mean?

Reply:
VM and native compute.

Reply:
> Sure it would seem like something sensible to document. I suggest doing it in another PR, and indeed `INSTALL` seems like the reasonable place for this.

I honestly don't know what to require in the general case. Thus, I just add the requirement for IEEE-754 arch in INSTALL in the current PR.